lib/logstash/filters/resolv_patch.rb

```diff
@@ -0,0 +1,29 @@
+require "resolv"
+
+# ref: https://github.com/logstash-plugins/logstash-filter-dns/issues/40
+#
+# JRuby versions prior to 9.1.16.0 have a bug which crashes IP address
+# the resolver after 64k unique IP addresses resolutions.
+
+version = JRUBY_VERSION.split(".").map(&:to_i)
+
+if version[0] == 9 && version[1] == 1 && version[2] < 16
```

Comment:
ruby's most common way to compare versions is through Gem::Version, we could rewrite:

```ruby
version = JRUBY_VERSION.split(".").map(&:to_i)

if version[0] == 9 && version[1] == 1 && version[2] < 16
```

as:

```ruby
if Gem::Version.new("9.1.16.0") > Gem::Version.new(JRUBY_VERSION)
```

or 
```ruby
RESOLV_PATCH_CONSTRAINT = Gem::Requirement.new("< 9.1.16.0")

if RESOLV_PATCH_CONSTRAINT.satisfied_by?(Gem::Version.new(JRUBY_VERSION))
```

Reply:
Does this also work on JRuby 1.7 as well? Have you checked if there has been much changes to resolv.rb since then?

I'm just wondering if the exact same patch will work for Logstash 5.x / JRuby 1.7.

Reply:
I stress tested the 1.7.25 and 1.7.27 (the 2 JRuby vesions used across the LS 5 versions) resolver and they are not affected by this bug. Only 9k need patching and 9.1.13.0 is the first 9k version we used.